client/rpc-api/src/state/mod.rs

```diff
@@ -136,4 +136,106 @@ pub trait StateApi<Hash> {
 	fn unsubscribe_storage(
 		&self, metadata: Option<Self::Metadata>, id: SubscriptionId
 	) -> RpcResult<bool>;
+
+	/// The `state_traceBlock` RPC provides a way to trace the re-execution of a single
+	/// block, collecting Spans and Events from both the client and the relevant WASM runtime.
+	/// The Spans and Events are conceptually equivalent to those from the [Tracing][1] crate.
+	///
+	/// The structure of the traces follows that of the block execution pipeline, so meaningful
+	/// interpretation of the traces requires an understanding of the Substrate chain's block
+	/// execution.
+	///
+	/// [Link to conceptual map of trace structure for Polkadot and Kusama block execution.][2]
+	///
+	/// [1]: https://crates.io/crates/tracing
+	/// [2]:
+	/// https://github.com/paritytech/substrate/tree/master/client/rpc-api/TraceBlock_Trace_Hierarchy.png
+	///
+	/// ## Node requirements
+	///
+	/// - Fully synced archive node.
+	/// - [Tracing enabled WASM runtimes](#creating-tracing-enabled-wasm-runtimes) for all runtime versions
+	/// that tracing is desired.
+	///
+	/// ## Node recommendations
+	///
+	/// - Use fast SSD disk storage.
+	/// - Run node flags to increase DB read speed (i.e. `--state-cache-size`, `--db-cache`).
+	///
+	/// ## Creating tracing enabled WASM runtimes
+	///
+	/// - Checkout commit of chain version to compile with WASM traces
+	/// - [diener](https://crates.io/crates/diener) can help to peg commit of substrate
```

Comment:
why is this needed? Should be merged into master then you shouldn't need to use diener?

Reply:
The user will need to either create or get ahold of tracing enabled wasm binaries since by default they are not tracing enabled. These instructions explain how to create tracing enabled wasm binaries - which I don't think we would ever check into source control for substrate.